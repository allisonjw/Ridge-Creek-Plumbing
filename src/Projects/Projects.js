import React from 'react';
import './Projects.scss';

export const Projects = ({ name, image, location, project_type }) => {

  return (
    <section className="portfolio_main-projects">
      <article className="project">
        <div 
          className="reveal-content" 
          aria-label="holds image that is revealed on hover">
              <img 
                src={image}
                alt={name}
                className="project_reveal-img">
              </img>
        </div>
        <div 
          className="reveal-top" 
          aria-label="name of project that is displayed when not being hovered">
          <div 
            className="reveal-content caption" aria-label="title of project that is displayed on page load">
              <h4 className="project_reveal-h4-name">{name}</h4>
              <h4 className="project_reveal-h4-location">{location}</h4>
          </div>
        </div>
      </article>
    </section>
  )
};

export default Projects;