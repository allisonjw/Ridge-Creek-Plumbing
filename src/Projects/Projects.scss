@import '../SCSS/variables';
@import '../SCSS/media-queries';

.project {
    display: table-cell;
    float: left;
    width: 301px;
    height: 197px;
    margin-right: 10px;
    margin-bottom: 10px;
    position: relative;
    border: 2px solid #005798;
    cursor: pointer;
    overflow: hidden;
}

.project:hover .reveal-top {
    top: -100%;
}
  
.reveal-content {
    box-shadow: 1px 1px 3px rgba(0,0,0,0.3) inset;
    // padding: 100px;
    color: #000;
    transition: left 0.4 ease;
}

.project_reveal-img {
    background-size: cover;
    padding: none;
    float: left;
    width: 301px;
    height: 197px;
}
  
.reveal-top {
    background: rgba(0,0,0,0.5);
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    transition: all 0.6s ease;
}
  
.caption {
    background: #0000008f;
    color: #fff;
}

.project_reveal-h4-name,
.project_reveal-h4-location{
    font-size: 1.4em;
    color: white;
    position: absolute;
    width: 100%;
    height: 100%;
    text-align: center;
    padding: 3%;
}
.project_reveal-h4-location {
    top: 35px;
}

