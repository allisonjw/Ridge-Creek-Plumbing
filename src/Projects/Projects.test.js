import React from 'react';
import Projects from './Projects';
import { shallow } from 'enzyme';

describe('Projects', () => {
    it('should match the snapshot', () => {
        let project_type = 'commercial';
        let name = 'Puyallup School District';
        let location = 'Puyallup, WA';
        let image = '../images/puyallup.jpg';

        const projectWrapper = shallow(
            <Projects 
                project_type={project_type}
                name={name}
                location={location}
                image={image}
            />
        );

    expect(projectWrapper).toMatchSnapshot();
  });
});