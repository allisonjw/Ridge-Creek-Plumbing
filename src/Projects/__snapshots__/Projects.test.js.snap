// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should match the snapshot 1`] = `
<section
  className="portfolio_main-projects"
>
  <article
    className="project"
  >
    <div
      aria-label="holds image that is revealed on hover"
      className="reveal-content"
    >
      <img
        alt="Puyallup School District"
        className="project_reveal-img"
        src="../images/puyallup.jpg"
      />
    </div>
    <div
      aria-label="name of project that is displayed when not being hovered"
      className="reveal-top"
    >
      <div
        aria-label="title of project that is displayed on page load"
        className="reveal-content caption"
      >
        <h4
          className="project_reveal-h4-name"
        >
          Puyallup School District
        </h4>
        <h4
          className="project_reveal-h4-location"
        >
          Puyallup, WA
        </h4>
      </div>
    </div>
  </article>
</section>
`;
